doc/dependency-resolution.md

```diff
@@ -1,356 +1,129 @@
-## Dependency Resolution
+# Dependency Resolution
 
-**Problem Statement**
+Dependency Resolution is a concept within the Publishing API to describe two
+processes:
 
-We're undergoing a migration to the new publishing platform. The publishing
-platform doesn't currently prescribe how the following features should work:
+- Determining the links that are presented with a content item when it is
+  represented in the Content Store.
+- Determining which content items require updating as a result of a change to
+  a different content item.
```

Comment:
I would normally phrase this as "re-presenting to the content store", rather than just updating. It also might be good to specify that this should only change the expanded links?